packages/core/test/oracle/SpamGuard.js

```diff
@@ -414,6 +414,61 @@ describe("SpamGuard", function () {
     assert.equal(statuses[3].status, "2"); // spam deletion request. resolved.
     assert.equal(statuses[4].status, "1"); // valid request. now active in voting.
   });
+
+  it("Correct bonds refund", async function () {
+    // Construct a simple price request that is valid and should not be considered as spam.
+    const time = Number(await voting.methods.getCurrentTime().call()) - 10; // 10 seconds in the past.
+
+    // Request prices and move to the next round where that will be voted on.
+    await voting.methods.requestPrice(validIdentifier, time).send({ from: registeredContract });
+    await voting.methods.requestPrice(spamIdentifier, time).send({ from: registeredContract });
+
+    // Construct the request to delete the spam votes.
+    await voting.methods.signalRequestsAsSpamForDeletion([[1, 1]]).send({ from: account1 });
+    const signalDeleteTime = await voting.methods.getCurrentTime().call();
+
+    // The caller of this method should have lost 10k votingTokens as a bond when calling this function.
+    assert.equal(await votingToken.methods.balanceOf(account1).call(), toWei("9990000")); // 10mm - 10k bond
+
+    // All the requests should be enqueued in the following voting round.
+    await moveToNextRound(voting, accounts[0]);
+
+    const spamDeleteIdentifier = padRight(utf8ToHex("SpamDeletionProposal 0"), 64);
+
+    // In the meantime update the spamDeletionProposalBond to 500 bond
+    const tx = await voting.methods.setSpamDeletionProposalBond(toWei("0.5")).send({ from: account1 });
+    assert.equal(tx.events.SpamDeletionProposalBondChanged.returnValues.newBond, toWei("0.5"));
+
+    // Commit votes. Vote on the valid request and vote on spam deletion request. Voting to delete the spam is with a
+    // vote of "1e18". Dont vote on the spam requests. Only vote from account2 to simplify the test.
+    const account = account2;
+    const roundId = (await voting.methods.getCurrentRoundId().call()).toString();
+    const salt = getRandomSignedInt();
+
+    const hash2 = computeVoteHash({
+      price: toWei("1"),
+      salt,
+      account,
+      roundId,
+      time: signalDeleteTime,
+      identifier: spamDeleteIdentifier,
+    });
+    await voting.methods.commitVote(spamDeleteIdentifier, signalDeleteTime, hash2).send({ from: account2 });
+
+    // Reveal the vote.
+    await moveToNextPhase(voting, accounts[0]);
+
+    await voting.methods.revealVote(spamDeleteIdentifier, signalDeleteTime, toWei("1"), salt).send({ from: account2 });
+
+    // Execute the spam deletion call.
+    await moveToNextRound(voting, accounts[0]);
+
+    await voting.methods.executeSpamDeletion(0).send({ from: account1 });
+
+    // Even if the bond changed in the meantime the user get's the original bond paid back.
+    assert.equal(await votingToken.methods.balanceOf(account1).call(), toWei("10000000")); // 10mm.
```

Comment:
nit: can we add an assert that the caller's balance also ends at the expected level?